Added on final step for verification project data

kkatusic opened the pull request:
@RamRamez I added additional step for submiting verifiacation. Didn't find any test related to this part of code, correct me if I'm wrong. 

Please can you check code and let me know, thx ;)

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

- **New Features**
  - Improved project verification submission process with enhanced handling for terms and conditions, and managing funds.

- **Tests**
  - Updated test cases to reflect changes in project verification statuses and added a new test for successful form update upon completion of all steps.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

kkatusic committed:
added on final step for verification project data

RamRamez commented on src/services/projectVerificationFormService.ts:
```diff
@@ -153,6 +153,15 @@ export const updateProjectVerificationFormByUser = async (params: {
+      updatedProjectVerificationForm =
+        await updateTermsAndConditionsOfProjectVerification({
+          projectVerificationId,
+          isTermAndConditionsAccepted: true,
```
I think you shouldn't change isTermAndConditionsAccepted state on SUBMIT step. If user has not accepted Term And Conditions, we shouldn't change that to true when PROJECT_VERIFICATION_STEPS.SUBMIT is called.

kkatusic replied:
He can't submit last step if all steps aren't checked, but can I somehow make some update here because I need `updatedProjectVerificationForm` object for code below?

RamRamez replied:
@kkatusic I think you can remove `updateTermsAndConditionsOfProjectVerification` and add below code to fetch projectVerificationForm:

```
  const updatedProjectVerificationForm = await findProjectVerificationFormById(
    projectVerificationId,
  );
```

kkatusic replied:
let me try

kkatusic replied:
@RamRamez when you get time please check, thx ;)

RamRamez replied:
Thanks @kkatusic. Approved!

I think there's a typo in `getdProjectVerificationData`

kkatusic committed:
fixed last step verification

RamRamez reviewed (approved):
Thanks @kkatusic. Approved!

I think there's a typo in `getdProjectVerificationData`

kkatusic committed:
fixed typo

kkatusic commented:
> Thanks @kkatusic. Approved!
> 
> I think there's a typo in `getdProjectVerificationData`

@RamRamez you are right, fixed, thx you for all help

RamRamez reviewed (approved):
Thanks @kkatusic

kkatusic commented:
> Thanks @kkatusic

@RamRamez it's not passing test, can you see error?

RamRamez commented:
@kkatusic lots of tests are failing, I think it's not related to your PR because staging is also failing.
@mohammadranjbarz do you know what's the problem?

mohammadranjbarz committed:
Merge branch 'staging' into add_ver_step

kkatusic commented:
@RamRamez please when you get time to review this, thx

RamRamez committed:
Merge branch 'staging' into add_ver_step

RamRamez reviewed (approved):
Thanks @kkatusic

RamRamez commented:
@kkatusic Some tests related to project verification are failing. Could you check please?

kkatusic commented:
> @kkatusic Some tests related to project verification are failing. Could you check please?

I checked here: https://github.com/Giveth/impact-graph/actions/runs/9827260425/job/27129764379?pr=1658 but I don't see any test failed? Am I looking at wrong report? 

As I mentioned before, I didn't find any test related to verification form, can you direct me in which file it is? thx @RamRamez

RamRamez commented:
@kkatusic If you click `test` tab and scroll down on that page, you can which tests are failing.
<img width="1312" alt="image" src="https://github.com/Giveth/impact-graph/assets/73441857/76c1ea78-011c-4cb4-8475-d39c7fa441fe">

kkatusic committed:
Updated project verification form resolver test

fixed previos 3 test with terms and condition update and added one last step test when user complete form and clicked "FINISH" button
